pages/api/checkout_sessions_supporting_sponsored.ts

```diff
@@ -0,0 +1,56 @@
+import { api } from "app/blitz-server"
+import { NextApiRequest, NextApiResponse } from "next"
+const stripe = require("stripe")(process.env.STRIPE_SECRET_KEY, { apiVersion: "2020-03-02" })
+
+const CreateSessionSupporting = async (req: NextApiRequest, res: NextApiResponse) => {
+  if (!req.query.email || !req.query.price_id) {
+    res.status(500).end("Incomplete request")
+  } else {
+    if (req.method === "POST" && req.query.price_id && req.query.email) {
+      try {
+        const coupon = await stripe.coupons.create({
+          duration: "forever",
```

Comment:
I want to double-check: Does the coupon expire as soon as it's used? "Forever" does not mean that it can be used repeatedly?